idaes/core/scaling/custom_scaler_base.py

```diff
@@ -541,16 +632,36 @@ def propagate_state_scaling(
             None
         """
         for bidx, target_data in target_state.items():
-            target_vars = target_data.define_state_vars()
-            source_vars = source_state[bidx].define_state_vars()
-
-            for state, var in target_vars.items():
-                for vidx, vardata in var.items():
-                    self.scale_variable_by_component(
-                        target_variable=vardata,
-                        scaling_component=source_vars[state][vidx],
-                        overwrite=overwrite,
-                    )
+            self.propagate_state_data_scaling(
+                target_state_data=target_data,
+                source_state_data=source_state[bidx],
+                overwrite=overwrite,
+            )
+
+    def propagate_state_data_scaling(
+        self, target_state_data, source_state_data, overwrite: bool = False
+    ):
+        """
+        Propagate scaling of state variables from one StateBlockData to another.
+
+        Args:
+            target_state_data: StateBlockData to set scaling factors on
+            source_state_data: StateBlockData to use as source for scaling factors
+            overwrite: whether to overwrite existing scaling factors
+
+        Returns:
+            None
+        """
+        target_vars = target_state_data.define_state_vars()
+        source_vars = source_state_data.define_state_vars()
+
+        for state, var in target_vars.items():
+            for vidx, vardata in var.items():
+                self.scale_variable_by_component(
+                    target_variable=vardata,
+                    scaling_component=source_vars[state][vidx],
+                    overwrite=overwrite,
+                )
```

Comment:
Similar point with this - is there a simple way to combine this with the existing function?

Reply:
Is the proposed behavior to have `propagate_state_scaling` check whether the inputs are indexed or unindexed, then handle those cases appropriately?